lib/src/main/java/io/ably/lib/rest/Auth.java

```diff
@@ -827,13 +829,31 @@ public AuthOptions getAuthOptions() {
      * Renew auth credentials.
      * Will obtain a new token, even if we already have an apparently valid one.
      * Authorization will use the parameters supplied on construction.
+     * @deprecated Because the method returns early before renew() completes and does not provide a completion
+     * handler for callers.
+     * Please use {@link Auth#renewAuth} instead
      */
+    @Deprecated
     public TokenDetails renew() throws AblyException {
         TokenDetails tokenDetails = assertValidToken(this.tokenParams, this.authOptions, true);
         ably.onAuthUpdated(tokenDetails.token, false);
         return tokenDetails;
     }
 
+    /**
+     * Renew auth credentials.
+     * Will obtain a new token, even if we already have an apparently valid one.
+     * Authorization will use the parameters supplied on construction.
+     *
+     * @return
+     * A single entry that contain a token detail and a future that represent an asynchronous result
+     * Clients must wait for the future result to finish before processing.
+     */
+    public Map.Entry<TokenDetails, Future<Void>> renewAuth() throws AblyException {
```

Comment:
Since this is our public API, do we really want to return this as a `Map.Entry`? Then the users would have to call `renewAuth().getKey()` to get the `TokenDetails` and `renewAuth().getValue()` to get the `Future`, which is not very intuitive IMO :thinking: Maybe let's create a new class that would act as a container for those two values? Something like `RenewAuthResult`? :thinking: